### PR TITLE
Stabilization of TestSimpleMetrics on CI

### PR DESCRIPTION
**Pull request by denis-tingajkin**
Signed-off-by: denis-tingajkin <denis.tingajkin@xored.com>

## Description
#1029
**Ready for reivew/merge**

## How Has This Been Tested?

- [X] Covered by existing integration testing
- [X] Added integration testing to cover
- [X] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: dataplane/vppagent/pkg/vppagent/vppagent.go [hunk 1]
```diff
@@ -323,11 +325,38 @@ func (v *VPPAgent) Init(common *common.DataplaneConfigBase, monitor *crossconnec
...
 	return nil
 }
 
+func (v *VPPAgent) setupMetricsCollector(monitor *crossconnect_monitor.CrossConnectMonitor) {
+	val, ok := os.LookupEnv(DataplaneMetricsCollectorEnabled)
+	if !ok {
+		logrus.Infof("%v not set, metrics collector is not using", DataplaneMetricsCollectorEnabled)
+		return
+	}
+	metricsCollectorEnabled, err := strconv.ParseBool(val)
```
Comment (haiodo) [inline dataplane/vppagent/pkg/vppagent/vppagent.go]: By default it should be enabled, and please disable by default options in SetupNodes in our tests.

File: test/integration/basic_monitor_crossconnect_metrics_test.go [hunk 2]
```diff
@@ -21,21 +24,26 @@ func TestSimpleMetrics(t *testing.T) {
...
+	requestPeriod := time.Second
+	variables := map[string]string{
+		vppagent.DataplaneMetricsCollectorEnabled:       "true",
+		vppagent.DataplaneMetricsCollectorRequestPeriod: requestPeriod.String(),
+	}
 
-	nodes := nsmd_test_utils.SetupNodes(k8s, nodesCount, defaultTimeout)
-	nsmd_test_utils.DeployICMP(k8s, nodes[nodesCount-1].Node, "icmp-responder-nse-1", defaultTimeout)
-	nsc := nsmd_test_utils.DeployNSC(k8s, nodes[0].Node, "nsc1", defaultTimeout)
+	nodes := nsmd_test_utils.SetupNodesConfig(k8s, nodesCount, defaultTimeout, []*pods.NSMgrPodConfig{
```
Comment (haiodo) [inline test/integration/basic_monitor_crossconnect_metrics_test.go]: Please also use default values for NSmgr.

File: dataplane/vppagent/pkg/vppagent/vppagent.go (hunk 1 above)
Comment (denis-tingajkin) [reply dataplane/vppagent/pkg/vppagent/vppagent.go]: thanks, fixed

File: test/integration/basic_monitor_crossconnect_metrics_test.go (hunk 2 above)
Comment (denis-tingajkin) [reply test/integration/basic_monitor_crossconnect_metrics_test.go]: thanks, fixed

File: dataplane/vppagent/pkg/vppagent/vppagent.go [hunk 3]
```diff
@@ -323,11 +325,37 @@ func (v *VPPAgent) Init(common *common.DataplaneConfigBase, monitor *crossconnec
...
+	v.setupMetricsCollector(monitor)
 	return nil
 }
 
+func (v *VPPAgent) setupMetricsCollector(monitor *crossconnect_monitor.CrossConnectMonitor) {
+	val, ok := os.LookupEnv(DataplaneMetricsCollectorEnabledKey)
+	if ok {
+		enabled, err := strconv.ParseBool(val)
+		if err != nil {
+			logrus.Errorf("Metrics collector is not using, %v ", err)
```
Comment (haiodo) [inline dataplane/vppagent/pkg/vppagent/vppagent.go]: I suppose we need to enable it in this case.

Comment (denis-tingajkin) [reply dataplane/vppagent/pkg/vppagent/vppagent.go]: fixed

**Pull request merged**